client/ayon_deadline/plugins/publish/nuke/submit_nuke_deadline.py

```diff
@@ -362,36 +363,24 @@ def payload_submit(
 
         # Include critical environment variables with submission
         keys = [
-            "PYTHONPATH",
-            "PATH",
-            "AYON_BUNDLE_NAME",
-            "AYON_DEFAULT_SETTINGS_VARIANT",
-            "AYON_PROJECT_NAME",
-            "AYON_FOLDER_PATH",
-            "AYON_TASK_NAME",
-            "AYON_APP_NAME",
-            "FTRACK_API_KEY",
-            "FTRACK_API_USER",
-            "FTRACK_SERVER",
-            "PYBLISHPLUGINPATH",
             "NUKE_PATH",
-            "TOOL_ENV",
-            "FOUNDRY_LICENSE",
-            "OPENPYPE_SG_USER",
+            "FOUNDRY_LICENSE"
         ]
 
         # add allowed keys from preset if any
         if self.env_allowed_keys:
             keys += self.env_allowed_keys
 
-        environment = {
+        nuke_specific_env = {
             key: os.environ[key]
             for key in keys
             if key in os.environ
         }
 
-        # to recognize render jobs
-        environment["AYON_RENDER_JOB"] = "1"
+        # Set job environment variables
+        environment = get_instance_job_envs(instance)
+        environment.update(get_ayon_render_job_envs())
+        environment.update(nuke_specific_env)
 
         # finally search replace in values of any key
         if self.env_search_replace_values:
```

Comment:
@BigRoy, the only case I can think of where we needed to include the PATH or PYTHONPATH variable was due to `self.env_search_replace_values`. I remember, a long time ago, we faced an issue where a client needed to distribute a specific locked machine license. However, I assume this is just an edge case, and we should proceed with merging your solution.

  

I'm currently testing it, and approval will come soon.

Reply:
Agree - removed.